### PR TITLE
reset output colors before exiting when interrupt encountered

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

**Comment by CaptainStabs** [comment]
Related to issue #1838

**Comment by CaptainStabs** [comment]
Just tested, works on windows, but with a severe loss of responsivity (Nearly 16 seconds after SIGINT is first pressed)

File: go/cmd/dolt/cli/stdio.go [hunk 1]
```diff
@@ -19,13 +19,27 @@ import (
...
 
+var outputClosed uint64
+
+func CloseOutput() {
+	if atomic.CompareAndSwapUint64(&outputClosed, 0, 1) {
+		fmt.Fprintln(CliOut)
+	}
+}
+
+func outputIsCLosed() bool {
```
Comment (zachmu) [inline go/cmd/dolt/cli/stdio.go:38]: CLosed => Closed

**Review by zachmu — APPROVED** [review]
LGTM

**Comment by bheni** [comment]
@CaptainStabs I've merged this to master.

In order to write colored output to the command line, we use [ansi escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code).  These codes are interpreted by the terminal window that renders the output.  When an ansi escape code is encountered the terminal window will change the color of the text that is written until more ansi escape codes are encountered changing the color to something new, or back to the default.

As Dolt writes text to standard out it is buffered.  Depending on the os and setup there could be a lot of text buffered to be written to standard out, and at the point that ctrl-c is hit.  If the program terminates immediately you are left in a state where your terminal window may still be actively writing colored text.  This change prevents further text from being written to standard out, and writes the ansi control codes to change the text back to the default.  The issue is that for these ansi control codes to be interpreted all the text that's already been buffered needs to be flushed.  That can lead to it feeling less responsive when you hit ctrl-c as it flushes and resets the text color.

With this change you have to options:
 1. Send sigint (ctrl+c) and know that your terminal colors will be reset but it may take a little time to flush
 2. Send sigterm (pkill / kill command in linux, not sure in windows) to kill dolt immediately, but your text color may not be properly reset as your terminal window will be left writing whatever color it wrote last.

**Pull request merged**